rcl_logging_noop/CMakeLists.txt

```diff
@@ -25,7 +25,7 @@ add_library(${PROJECT_NAME} src/rcl_logging_noop.cpp)
 target_link_libraries(${PROJECT_NAME} PRIVATE
   rcutils::rcutils)
 target_link_libraries(${PROJECT_NAME} PUBLIC
-  rcl_logging_interface::rcl_logging_interface)
+  $<BUILD_INTERFACE:rcl_logging_interface::rcl_logging_interface>)
```

Comment:
So I'm honestly not sure about all of this, but I don't know that this is correct.  I feel like this will link to the *build* directory, not the install directory, and that doesn't seem correct to me.  Maybe @sloretz can chime in here to give a bit more explanation.

Reply:
If I may, `$<BUILD_INTERFACE:...>` means that the property will be removed when the target is installed via `install(EXPORT)` but available when `export()` is used or when the target is consumed from within the build-tree. ([doc](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:BUILD_INTERFACE)) So the installed library will link to nothing instead of a library that it doesn't bring in (as part of #102)